lib/rules/jquery-dollar-sign-reference.js

```diff
@@ -126,15 +126,34 @@ module.exports = {
       };
     }
 
+    function getValueNodes(node) {
+      var results = [];
+      if (node.type === 'LogicalExpression') {
+        results.push.apply(results, getValueNodes(node.left));
+        results.push.apply(results, getValueNodes(node.right));
+      } else if (node.type === 'ConditionalExpression') {
+        results.push.apply(results, getValueNodes(node.consequent));
+        results.push.apply(results, getValueNodes(node.alternate));
+      } else {
+        results.push(node);
+      }
+      return results;
+    }
+
     function checkForValidjQueryReference(node, left, right) {
       if (right == null) { return; }
       var isjQueryRef = isjQueryReference(left);
-      var isjQueryVal = isjQueryValue(right);
-      // console.log(right);
+      var valueNodeTypes = getValueNodes(right).map(isjQueryValue);
+      var hasDefinitejQueryValue = valueNodeTypes.some(function(nodeType) {
```

Comment:
This could be a preference thing, but what's our stance on utilizing arrow functions?
```
var hasDefinitejQueryValue = valueNodeTypes.some(nodeType => nodeType.definite);
var hasRegularValue = valueNodeTypes.some(nodeType => !nodeType.possible ;
```